src/object/sound_object.cpp

```diff
@@ -0,0 +1,164 @@
+//  SuperTux
+//  Copyright (C) 2023 mrkubax10 <mrkubax10@onet.pl>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "object/sound_object.hpp"
+
+#include <limits>
+
+#include "audio/sound_manager.hpp"
+#include "audio/sound_source.hpp"
+#include "editor/editor.hpp"
+#include "util/reader_mapping.hpp"
+
+SoundObject::SoundObject(const ReaderMapping& mapping) :
+  GameObject(mapping),
+  ExposedObject<SoundObject, scripting::SoundObject>(this),
+  m_sample(),
+  m_sound_source(),
+  m_volume(),
+  m_play_interval(),
+  m_delay(0.0f),
+  m_playing(false),
+  m_scheduled_for_removal(false)
+{
+  mapping.get("sample", m_sample, "");
+  mapping.get("volume", m_volume, 1.0f);
+  mapping.get("play-interval", m_play_interval, 0.0f);
+
+  prepare_sound_source();
+}
+
+SoundObject::SoundObject(float vol, float play_interval, const std::string& file) :
+  ExposedObject<SoundObject, scripting::SoundObject>(this),
+  m_sample(file),
+  m_sound_source(),
+  m_volume(vol),
+  m_play_interval(play_interval),
+  m_delay(0.0f),
+  m_playing(false),
+  m_scheduled_for_removal(false)
+{
+  prepare_sound_source();
+}
+
+SoundObject::~SoundObject()
+{
+  stop_looping_sounds();
+}
+
+void
+SoundObject::update(float dt_sec)
+{
+  if (m_scheduled_for_removal)
+  {
+    remove_me();
+    return;
+  }
```

Comment:
`remove_me()` could be called directly on the object, whenever it should be scheduled for removal. `GameObjectManager` will take care of it.